pkg/securitycontextconstraints/sccmatching/matcher.go

```diff
@@ -77,7 +78,7 @@ func authorizedForSCC(sccName string, info user.Info, namespace string, a author
 		Resource:        "securitycontextconstraints",
 		ResourceRequest: true,
 	}
-	decision, reason, err := a.Authorize(attr)
+	decision, reason, err := a.Authorize(context.TODO(), attr)
```

Comment:
I think we should allow the caller to pass a `context`. We can change the current interface from
```go
type SCCMatcher interface {
	FindApplicableSCCs(namespace string, user ...user.Info) ([]*securityv1.SecurityContextConstraints, error)
}
```
to 
```go
type SCCMatcher interface {
	FindApplicableSCCs(ctx context.Context, namespace string, user ...user.Info) ([]*securityv1.SecurityContextConstraints, error)
}
```
Let me know if I should proceed with the change. This would be a breaking change and we expect the the user of this package to fix their code to use the new version or do we need to keep the above interface backwards compatible?

Reply:
yes, please update the interface. The expectations are that the users of this package will adjust.

Reply:
where is the interface used? If it is only local, then it definitely makes sense.

Reply:
`SCCMatcher` is a public interface and `NewDefaultSCCMatcher` function returns an instance of it. I can see it being used by other packages in the same repo (`apiserver-library-go` repo).  Other go packages in different could be using it too since it is public.

Reply:
openshift-apiserver and origin's kube-apiserver might. Please double check.

Reply:
`openshift/openshift-apiserver` uses it. `openshift/kubernetes-apiserver` does not use it. @sttts